src/main_menu.cpp

```diff
@@ -1190,6 +1185,10 @@ bool main_menu::load_character_tab( bool transfer )
                     }
 
                     if( g->load( savegames[sel3] ) ) {
+                        on_out_of_scope cleanup( []() {
+                            g->u = avatar();
+                            world_generator->set_active_world( nullptr );
+                        } );
```

Comment:
This would change the behavior of the code. For example, if `g->setup()` fails and `continue` is executed, `cleanup` gets deconstructed and `world_generator->set_active_world( nullptr )` is called, ensuring that the active world is reset on error. But with this change it does nothing and the active world remains whatever erroneous world that was previous set.